core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala

```diff
@@ -208,9 +210,9 @@ class R2dbcExecutor(val connectionFactory: ConnectionFactory, log: Logger)(impli
     }
   }
 
-  def selectOne[A](logPrefix: String)(statement: Connection => Statement, mapRow: Row => A): Future[Option[A]] = {
-    select(logPrefix)(statement, mapRow).map(_.headOption)
-  }
+  def selectOne[A](logPrefix: String)(statement: Connection => Statement, mapRow: Row => A): Future[Option[A]] =
+    // Suggestion: if mapRow returns `null`, we want to make it a None by default
```

Comment:
in that case we should use `null` as a filter also for mapRow in `select`, and then this wouldn't need to change?

Reply:
Good point, the `select` could discard null values. That makes most sense when selecting a single value. Anyway, the `mapRow` defines what is `null`. Can be a that a whole `Row` is mapped to `null`. 

Another question is if we want to make it a general purpose public API for users. They could benefit of it when building projections, for example. Don't need to be a full blow framework around R2DBC, but some basic building blocks to make R2DBC play nice with Akka and Akka Streams. 

If we don't want that, those methods can be limited to the things we need internally.